File: src/leetQ61/Solution.java
package leetQ61;

/**
 * Created by zhangzhongwen on 2019/1/12
 */
public class Solution {
    public static void main(String[] args) {
        ListNode node1 = new ListNode(1);
        ListNode node2 = new ListNode(2);
        ListNode node3 = new ListNode(3);
        ListNode node4 = new ListNode(4);
        ListNode node5 = new ListNode(5);

        node1.next = node2;
        node2.next = node3;
        node3.next = node4;
        node4.next = node5;
        ListNode listNode = rotateRight(node1, 2);
        while (listNode != null) {
            System.out.println("value : " + listNode.val);
            listNode = listNode.next;
        }
    }

    public static ListNode rotateRight(ListNode head, int k) {
        if (head == null || head.next == null) return head;
        ListNode last = head;
        int length = 1;
        while (last.next != null) {
            last = last.next;
            length++;
        }
        last.next = head;
        ListNode temp = head;
        int flag = length - (k % length);
        for (int i = 1; i < flag; i++) {
            temp = temp.next;
        }
        ListNode ans = temp.next;
        temp.next = null;
        return ans;
    }
}

File: src/leetQ61/Solution.kt
package leetQ61

/**
 * Created by zhangzhongwen on 2019/1/12
 */
fun main(args: Array<String>) {
    val node1 = ListNode(1)
    val node2 = ListNode(2)
    val node3 = ListNode(3)
    val node4 = ListNode(4)
    val node5 = ListNode(5)

    node1.next = node2
    node2.next = node3
    node3.next = node4
    node4.next = node5
    var listNode = rotateRight(node1, 2)
    while (listNode != null) {
        println("value : " + listNode.`val`)
        listNode = listNode.next
    }
}

fun rotateRight(head: ListNode?, k: Int): ListNode? {
    if (head?.next == null) return head
    var last: ListNode = head
    var length = 1
    while (last.next != null) {
        last = last.next
        length++
    }
    last.next = head
    var temp: ListNode = head
    val flag = length - k % length
    for (i in 1 until flag) {
        temp = temp.next
    }
    val ans = temp.next
    temp.next = null
    return ans
}